kernel/src/expressions/scalars.rs

```diff
@@ -222,6 +255,15 @@ impl Display for Scalar {
                 }
                 write!(f, ")")
             }
+            Self::Map(data) => {
+                write!(f, "{{")?;
+                let mut delim = "";
+                for (key, val) in &data.pairs {
+                    write!(f, "{delim}{key}: {val}")?;
```

Comment:
Probably easier to read as:
```suggestion
                    write!(f, "  {delim}{key}: {val}")?;
```

The other stuff doesn't indent though so maybe just stay consistent.

Reply:
hm I'm a bit torn: I generally agree this would be better in isolation but doing something different that every other one seems non-ideal - how about a follow up `good-first-issue` that could be adding indentation to all?

Reply:
I think I missed something here... `delim` changes to `", "` (with a trailing space) for all entries after the first. Why do we need another unconditional space here?

Reply:
doh, yep I was thinking this was newline delimited, but it's not.